### DISCUSSION TITLE
NightDriverDesktop early look

### DISCUSSION THREAD

**Discussion by davepl** [PlummersSoftwareLLC/NightDriverStrip#569] [General]
I've been tinkering with a desktop app to control server-side effects that are rendered on the PC and then sent to NightDriverStrip over wifi.  This is the setup I use at my house to control everything, but it's a console app.

So I've written a WinForms app that includes a listview detailing all of the client connections and an LED visualizer so that you can see what's supposed to be happening on the strips.

Pretty early, but here's a sneak peek at it!

![2023-12-15_13-59-22](https://github.com/PlummersSoftwareLLC/NightDriverStrip/assets/3698967/61be7875-391c-48e3-87b5-e56b6e354380)

**Comment by smilima** [comment]
Dave,

Looks great!  I have always used WinForms for desktop applications. I feel
there has been a major step backwards with Visual Studio not supporting
WinForms (or something similar) with C++. I don’t want to write a program
for my GUI using XML. I feel like I have to write a program for my program!

Andrew
